### ISSUE TITLE
query comparative source value lost on save

### ISSUE THREAD

**Issue by EvanLomas** [deldrid1/node-red-contrib-firebase#19]
When adding a query to a once() node, setting the comparative value source to "msg." does nothing as the value seems to revert to "AZ" upon re-opening.

**Comment by deldrid1** [comment]
Can you provide a flow as an example of the bug here?  From your description, its difficult to understand if you are using the nodes properly or if you've uncovered an actual issue.

**Comment by EvanLomas** [comment]
Here is the flow output. It stores valType correctly as "msg" , but when the node's editor pane is opened again it shows "AZ" instead of "msg."

```
[
    {
        "id": "4d881408.0c80bc",
        "type": "firebase.once",
        "z": "b992b582.b0ddf8",
        "name": "",
        "firebaseconfig": "",
        "childpath": "",
        "repeatifnull": false,
        "eventType": "value",
        "queries": [
            {
                "name": "orderByKey",
                "value": "something",
                "valType": "msg"
            }
        ],
        "x": 195.5,
        "y": 85,
        "wires": [
            []
        ]
    }
]
```

If you open and save the editor the valType is lost and it reverts back to "str"

```
[
    {
        "id": "a120e2cc.0759b",
        "type": "firebase.once",
        "z": "b992b582.b0ddf8",
        "name": "",
        "firebaseconfig": "",
        "childpath": "",
        "repeatifnull": false,
        "eventType": "value",
        "queries": [
            {
                "name": "orderByChild",
                "value": "something",
                "valType": "str"
            }
        ],
        "x": 194.5,
        "y": 99,
        "wires": [
            []
        ]
    }
]
```


node-red-contrib-firebase -> 1.1.1
node-red -> 1.16.2
Node.js -> 6.10.1

**Comment by ssoleman** [comment]
Here is a commit that fixes this issue @ 09ecf33ac803fba5ffdfd27df413fa2e71441199. Note that I am still working on this, therefore it is still buggy (such as the typedinput type not resizing correctly).
